pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -461,6 +461,13 @@ func ServerPreferredNamespacedGroupVersionResources(discoveryClient discovery.Di
 	}
 	for _, apiGroup := range serverGroupList.Groups {
 		preferredVersion := apiGroup.PreferredVersion
```

Comment:
I'd prefer for you to make a "GetPreferredVersion" function somewhere for reuse.